models/DataObject/Classificationstore/KeyGroupRelation/Dao.php

```diff
@@ -96,6 +96,6 @@ public function update(): void
             }
         }
 
-        Helper::insertOrUpdate($this->db, self::TABLE_NAME_RELATIONS, $data);
+        Helper::upsert($this->db, self::TABLE_NAME_RELATIONS, $data, ['groupId', 'keyId']);
```

Comment:
why not use `$this->getPrimaryKey()` here?

Reply:
Missed that one after changing the approach.